src/Razor/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/ChildComponent_Generic_Explicit_OverrideCascade/TestComponent.codegen.cs

```diff
@@ -0,0 +1,36 @@
+// <auto-generated/>
+#pragma warning disable 1591
+namespace Test
+{
+    #line hidden
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Components;
+    public partial class TestComponent : Microsoft.AspNetCore.Components.ComponentBase
+    {
+        #pragma warning disable 1998
+        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder __builder)
+        {
+            __builder.OpenComponent<Test.Grid<DateTime>>(0);
+            __builder.AddAttribute(1, "Items", Microsoft.AspNetCore.Components.CompilerServices.RuntimeHelpers.TypeCheck<System.Collections.Generic.IEnumerable<DateTime>>(
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+                                Array.Empty<DateTime>()
+
+#line default
+#line hidden
+#nullable disable
+            ));
+            __builder.AddAttribute(2, "ChildContent", (Microsoft.AspNetCore.Components.RenderFragment)((__builder2) => {
+                __builder2.OpenComponent<Test.Column<System.TimeZoneInfo>>(3);
```

Comment:
Question:

We could think this is an error. The experience here is the same as it is today, isn't it?

There's not even a guaranteed runtime error here since there's nothing is passed to the child component.

Reply:
Why would we think of this as an error? This test's `.razor` code was legal in the past and should remain legal (with the same meaning) after this PR.

The only purpose of this test case is to show that the new type inference logic takes a lower precedence than explicitly-specific type parameters.